### DISCUSSION TITLE
I want to use timeField in Forms

### DISCUSSION THREAD

**Discussion by s-ohnishi** [phalcon/cphalcon#15943] [Q&A] [answered]
Looking at the manual and github, there are some forms parts that exist in Phalcon\Tag but not in Phalcon\Forms\Element.
 (For example, type="time")
I would like to use Forms because it is convenient to set default values ​​and validation, but is there a way to use it indirectly?
Is it necessary to write value settings and validation separately from other elements just for those purposes? Most of the elements can be described in Forms, so I feel that it is very troublesome to prepare separately for process description and form management.
Maybe it's not implemented at the moment, but can we expect it to be implemented in the near future?

**Comment by niden** [comment]
That whole namespace (Forms/HTML Elements) is to be reworked at some point in the future. There has been some work done for v5 but it is not fully complete. The idea is that for the forms you would pass an Html Helper factory which will allow you to create any number of elements that are available.

If you want you can open an issue but note it will be addressed after v5 is out, possibly in v6

**Reply by s-ohnishi** [reply]
Thank you for your reply.
Please tell me a little more about "pass an Html Helper factory".
Or, if there is a sample page, I would like to know the URL.

**Comment by niden** [comment, marked as answer]
Have a look here:

https://github.com/phalcon/cphalcon/blob/5.0.x/phalcon/Forms/Element/AbstractElement.zep#L521

This is the `tag` service that is used by each element. In the future we will rework this so that it is automatically injected in the constructor of the element as well as the form itself.

For now it is used to generate the necessary HTML per element

What you can also do is create an element class yourself for time like this


```php

namespace Phalcon\Forms\Element;

use Phalcon\Tag;

/**
 * Phalcon\Forms\Element\Time
 *
 * Component INPUT[type=time] for forms
 */
class Time extends AbstractElement
{
    /**
     * @var string
     */
    protected $method = "inputTime";
}
```

`inputTime` is a service in the Html Tag Factory so it will generate the tag you need.

The way things are now we have one class for forms one for HTML and we need to merge those somehow or at least create elements such as the implementation above to keep those two namespaces in sync.

**Reply by s-ohnishi** [reply]
Thank you very much.
I will create my own class that inherits from Phalcon \ Forms \ Element \ AbstractElement.

**Reply by s-ohnishi** [reply]
I made a timefield class and passed it, but the following error was displayed.
It seems that the render () method is required, but how should this be implemented?
> TimeField contains 1 abstract method and must therefore be declared abstract or implement the remaining methods (Phalcon\Forms\Element\ElementInterface::render)

**Reply by s-ohnishi** [reply]
Added as follows
```
    public function render(array $attributes = []):string
    {
        return \Phalcon\Tag::timeField(
            $this->prepareAttributes($attributes)
        );
    }
```
Is this okay?

**Reply by niden** [reply]
For v4 that should be OK. For v5 look at my example above.

**Reply by s-ohnishi** [reply]
OK.
Thank you very much.
